viewing_party/main.py

```diff
@@ -0,0 +1,200 @@
+
+def watch_movie(user_data, title):
+    #Watchlist key value (list), will be stored in watchlist_list 
+    # and then iterated.
+    watchlist_list = user_data["watchlist"]
+    #Each element of that list is a movie represented with a 
+    # dictionary containing the keys: title, genre and rating. 
+    for movie in watchlist_list:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            watchlist_list.remove(movie)
+    return user_data
+
+
+#--------FUNCTIONS FOR WAVE02--------#
+
+def get_watched_avg_rating(user_data):
+    #Initialize a counter in 0
+    movies_rating = 0
+    movies_watched_list = user_data["watched"]
+    #If watched movies list is empty, return 0.0 as average
+    if len(movies_watched_list) == 0:
+        return 0.0
+    #If not empty, add each movie rating to the total (movies_rating)
+    for movie in movies_watched_list:
+        movies_rating += movie["rating"]
+    return movies_rating / len(movies_watched_list)
+
+'''
+This function creates a genre dictionary to store the movie's
+genre and how many times that gender has been watched.
+OUTPUT: Returns the most frequent genre wanched (string).
+'''
+def get_most_watched_genre(user_data):
+    #Initialize an empty dictionary, empty string and counter in 0
+    genre_dictionary = {}
+    most_frecuent_genre = ""
+    most_frecuent_genre_amount = 0
+    #Generate a list of movies watched by user
+    movies_watched_list = user_data["watched"]
+    #If this list is empty return None
+    if not len(movies_watched_list):
+        return None
+    #Loop though the movies
+    for movie in movies_watched_list:
+        #Add a genre to the dictionary or increment the value by 1
+        if movie["genre"] in genre_dictionary:
+            genre_dictionary[movie["genre"]] += 1
+        else:
+            genre_dictionary[movie["genre"]] = 1
+        #Compare the value of this genre with the most frequent
+        if genre_dictionary[movie["genre"]] > most_frecuent_genre_amount:
+            most_frecuent_genre_amount = genre_dictionary[movie["genre"]]
+            most_frecuent_genre = movie["genre"]
+    return most_frecuent_genre
+
+
+#--------FUNCTIONS FOR WAVE03--------#
+
+'''
+This function was created for easier analysis and understanding of future functions.
+OUTPUT: Return a dictonary where key are the titles of movies watched by 
+any friend of user, and value is another dictionary with movie's info.
+'''
+def get_friends_movies_watched_titles(friends_list):
+    #Initialize an empty dictionary
+    movies_watched = {}
+    for friend in friends_list:
+        #Generate a list of movies watched by this friend
+        movies_watched_list = friend["watched"]
+        for movie in movies_watched_list:
+            #Loop through movies and add each new title to the dictionary
+            if movie["title"] in movies_watched:
+                continue
+            else:
+                movies_watched[movie["title"]] = movie
```

Comment:
let's try to avoid using `continue` as much as possible. 

```suggestion
            if movie["title"] not in movies_watched:
                movies_watched[movie["title"]] = movie
```